### PR TITLE
Improvement: DO-3471 Added message actions to show for each message

### PR DESCRIPTION
**Pull request by jsumiguin-causalens**
## Motivation and Context

Added functionality to append actions to a message. Actions will be additional elements that will be rendered with the message if the user plans to add additional functionality to the message.

## Implementation Description

 - Added actions property to `Message`
 - Passed this to `Message` component to be rendered in the UI
 
## How Has This Been Tested?

Locally
## PR Checklist:

- [x] I have implemented all requirements? (see JIRA, project documentation).
- [x] I am not affecting someone else's work, If I am, they are included as a reviewer.
- [x] I have added relevant tests (unit, integration or regression).
- [x] I have added comments to all the bits that are hard to follow.
- [x] I have added/updated Documentation.
- [x] I have updated the appropriate changelog with a line for my changes.

### CODE REVIEW

File: packages/ui-components/src/chat/chat.tsx [hunk 1]
```diff
@@ -215,6 +217,7 @@ function Chat(props: ChatProps): JSX.Element {
                         onDelete={onDeleteMessage}
                         value={message}
                         isEditable={didUserWriteMessage(message, props.activeUser)}
+                        messageActions={message.actions}
```
Comment (krzysztof-causalens) [inline packages/ui-components/src/chat/chat.tsx:220]: The entire message is already passed in as `value` do we need the extra prop?

File: packages/ui-components/src/types.tsx [hunk 2]
```diff
@@ -79,6 +81,8 @@ export interface Message {
     updated_at: string;
     /** User data of the person who wrote the message */
     user: UserData;
+
+    actions?: React.ReactNode[];
```
Comment (krzysztof-causalens) [inline packages/ui-components/src/types.tsx:85]: Missing docstring

File: packages/ui-components/src/chat/chat.tsx [hunk 3]
```diff
@@ -104,6 +104,8 @@ export interface ChatProps extends InteractiveComponentProps<Message[]> {
     isPopup?: boolean;
     /** A component showing the loading state of the chat, it appears above the input area, when not loading the caller can set it to null */
     loadingComponent?: React.ReactNode;
+
+    /** Pass through of className property */
```
Comment (krzysztof-causalens) [inline packages/ui-components/src/chat/chat.tsx:108]: Leftover?

**Review by krzysztof-causalens — APPROVED** [review]
Looks good. Could you add an example message with action(s) to the storybook page for future reference of intended usage?

**Review by sam-causalens — APPROVED** [review]
Looks good once there's a storybook example 👍

**Pull request merged**